### PR TITLE
e2e: verify existence of App CRs

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/giantswarm/issues/5278

Check that App CRs for each component of Release CR created have been created.

### CODE REVIEW

File: integration/test/releasehandling/release_handling_test.go [hunk 1]
```diff
@@ -45,6 +45,8 @@ var releaseCR = &releasev1alpha1.Release{
 //	- Creates a Release CR.
 //	- Checks if the CR has "release-operator.giantswarm.io/release-cycle-phase: upcoming" label reconciled.
 //	- Checks if the CR has ".status.cycle.phase: upcoming" status reconciled.
+//	- Verify aws-operator.4.6.0 app exists.
```
Comment (kopiczko) [inline integration/test/releasehandling/release_handling_test.go]:
```suggestion
//	- Verifies App CRs for the Release CR components exist.
```

File: integration/test/releasehandling/release_handling_test.go [hunk 2]
```diff
@@ -45,6 +45,8 @@ var releaseCR = &releasev1alpha1.Release{
...
+//	- Verify cert-operator.0.1.0 app exists.
```
Comment (kopiczko) [inline integration/test/releasehandling/release_handling_test.go]:
```suggestion
```

File: integration/test/releasehandling/release_handling_test.go [hunk 3]
```diff
@@ -111,4 +113,40 @@ func TestReleaseHandling(t *testing.T) {
...
+
+			return nil
+		}
+		b := backoff.NewMaxRetries(30, 1*time.Second)
+
+		err := backoff.Retry(o, b)
+		if err != nil {
+			t.Fatalf("err == %v, want %v", err, nil)
+		}
+	}
```
Comment (kopiczko) [inline integration/test/releasehandling/release_handling_test.go]: Can we make it the same way as in `multireleasehandling`? That is a bit more compact.
https://github.com/giantswarm/release-operator/pull/107/files#diff-d335cb88aca0d0ed82d92beb2ae7f0e8R119

**Comment by TheoBrigitte** [comment]
e2e test is failing due to how `e2e-harness` compiles the binary, we are missing the version injection and therefore version = "n/a"

```
        /go/src/github.com/giantswarm/release-operator/pkg/controller/resource/app/create.go:26:
        App.application.giantswarm.io "aws-operator.4.6.0" is invalid: metadata.labels: Invalid value: "n/a": a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')
D 05/07 12:55:21 /apis/release.giantswarm.io/v1alpha1/releases/aws.v6.1.0 app reconciled object | operatorkit/controller/controller.go:295 | controller=release-operator-release | event=update | loop=3 | version=427
```

File: integration/test/releasehandling/release_handling_test.go (hunk 3 above)
Comment (TheoBrigitte) [reply integration/test/releasehandling/release_handling_test.go]: good shout. updated this

**Comment by TheoBrigitte** [comment]
Yes, version is used in label
https://github.com/giantswarm/release-operator/blob/9c0ae990378ab627361e259a4eee8e40a01d3580/service/controller/release/resource/app/desired.go#L101

**Comment by TheoBrigitte** [comment]
e2e test was fixed by updating e2e-harness: https://github.com/giantswarm/e2e-harness/pull/198

**Pull request merged**